# EQUIPO:
* Rodrigo Mendoza Rodriguez
* Gael Arturo Muñoz Delgado
* Fernando Leon Franco

# Contexto
Una empresa de venta de dispositivos electrónicos observa fluctuaciones importantes en la demanda mensual. La gerencia quiere mejorar la planificación de inventarios y solicita a los analistas realizar un pronóstico rápido utilizando datos recientes.
El área de compras debe decidir si incrementa o no el inventario del siguiente mes, pero la decisión implica un riesgo financiero. Su equipo es responsable de entregar un pronóstico razonado y una recomendación sobre la acción a tomar.

In [1]:
import pandas as pd
import numpy as np

tabla = pd.DataFrame({'Mes': ['Enero', 'Febrero', 'Marzo', 'Abril', 'Mayo'], 'Demanda': [120, 140, 135, 160, 170]})
tabla

,Mes,Demanda
0,Enero,120
1,Febrero,140
2,Marzo,135
3,Abril,160
4,Mayo,170


1. En los ultimos 2 meses se puede apreciar una alza en la demanda, lo que puede indicar un crecimiento en el mes de junio.

# CASO UTILIZANDO PROMEDIO MENSUAL

In [5]:
tabla['Variacion'] = tabla['Demanda'].pct_change() 

promedio_variacion = tabla['Variacion'].mean()

ultimo_valor = tabla['Demanda'].iloc[-1]
pronostico_junio = ultimo_valor * (1 + promedio_variacion)

print(f"La Variación Promedio Mensual (VPM) es: {promedio_variacion:.4f} ({promedio_variacion*100:.2f}%)")
print(f"El pronóstico para Junio es: {pronostico_junio:.2f} unidades")
tabla

La Variación Promedio Mensual (VPM) es: 0.0947 (9.47%)
El pronóstico para Junio es: 186.09 unidades


,Mes,Demanda,Promedio_Movil_3,Variacion
0,Enero,120,NaN,NaN
1,Febrero,140,NaN,0.166667
2,Marzo,135,131.666667,-0.035714
3,Abril,160,145.000000,0.185185
4,Mayo,170,155.000000,0.062500


Con base en su pronóstico:
1. Concluyan si la empresa debería aumentar inventario, mantenerlo o comprar solo lo necesario por riesgo de sobrestock. \
La empresa debería aumentar el inventario de manera importante. Al utilizar la Variación Promedio Mensual, el pronóstico para junio es de 186 unidades. Este método es el más adecuado porque, a diferencia del promedio móvil, sí reconoce la tendencia de crecimiento de los últimos meses y evita que nos quedemos cortos de producto.

2. Analicen el riesgo:
* ¿Qué ocurre si la demanda real es mayor al pronóstico? \
Perdera terreno frente a cometidores, clientes insastifechos y una gran perdida de ventas

* ¿Qué ocurre si es menor? \
Sobrestock, capital sin moverse y mayor costo de almacenamiento.

3. Redacten una recomendación breve para la gerencia justificando su decisión. \
Se recomienda realizar una compra de 186 unidades. La tendencia histórica reciente muestra una recuperación sólida y un crecimiento promedio cercano al 10% mensual. Prepararse con este volumen asegura cubrir la demanda proyectada y maximizar los ingresos ante el buen ritmo de ventas actual.